internal/os/os_test.go

```diff
@@ -0,0 +1,80 @@
+package os
+
+import (
+	"bytes"
+	"strings"
+	"testing"
+
+	root "github.com/equinix/metal-cli/internal/cli"
+	outputPkg "github.com/equinix/metal-cli/internal/outputs"
+	"github.com/spf13/cobra"
+)
+
+func TestCli_RegisterCommands(t *testing.T) {
+	subCommand := "operating-systems"
+	consumerToken := ""
+	apiURL := ""
+	Version := "metal"
+	rootClient := root.NewClient(consumerToken, apiURL, Version)
+	type fields struct {
+		MainCmd  *cobra.Command
+		Outputer outputPkg.Outputer
+	}
+	type args struct {
+		name string
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		want    *cobra.Command
+		cmdFunc func(*testing.T, *cobra.Command)
+	}{
+		{
+			name: "NoArgs",
+			fields: fields{
+				MainCmd:  &cobra.Command{},
+				Outputer: outputPkg.Outputer(&outputPkg.Standard{}),
+			},
+			args: args{name: "testCommand"},
+			want: &cobra.Command{},
+			cmdFunc: func(t *testing.T, c *cobra.Command) {
+				t.Helper()
+				root := c.Root()
+				root.SetArgs([]string{subCommand})
+				if err := root.Execute(); err != nil {
+					t.Logf("%+v", root.Args)
+					t.Error("expected an error")
+				}
+			},
+		},
+		{
+			name: "get",
+			fields: fields{
+				MainCmd:  NewClient(rootClient, outputPkg.Outputer(&outputPkg.Standard{})).NewCommand(),
+				Outputer: outputPkg.Outputer(&outputPkg.Standard{}),
+			},
+			args: args{name: "testCommand"},
+			want: &cobra.Command{},
+			cmdFunc: func(t *testing.T, c *cobra.Command) {
+				t.Helper()
+				root := c.Root()
+				out := &bytes.Buffer{}
+				root.SetArgs([]string{subCommand, "get"})
+				if err := root.Execute(); err != nil {
```

Comment:
I don't think the logic here is quite right, and I'm concerned that this test also appears to pass in CI.

The first check here will fail the test if the command succeeded; then on line 68 we check that the output includes a string that is displayed if the command succeeded.  Put together, that means the test can only succeed if it produced an error _and_ successfully displayed OS information.